README.md

```diff
@@ -171,7 +176,7 @@ See the [C++ quickstart](doc/quickstart/cpp.md) and the
 [examples](https://github.com/mlpack/examples) repository for some examples of
 mlpack applications in C++, with corresponding `Makefile`s.
 
-### 3.1. Including mlpack and improving compile time
+### 3.2. Improving compile time
```

Comment:
Up to you, but it might be worth updating the table of contents at the top with this subsection and the new one you added.  I like the shorter title you used here. :+1:

Reply:
Suggest to use "Reducing" instead of "Improving".   The latter is ambiguous.